### DISCUSSION TITLE
[argparse modification] - how to ignore unknown argparse options?

### DISCUSSION THREAD

**Discussion by Formartha** [pytest-dev/pytest#11793] [Q&A]
Hi,

We have a use case where we need to ignore unknown parser.addoption(xyz) to support backward compatibility of pipelines.
I didn't find any documentation to avoid failing the test framework if unknown option was added.

I would appreciate your help.

example of failure:

`** pytest_addoption - adding additional parameters **
ERROR: usage: _jb_pytest_runner.py [options] [file_or_dir] [file_or_dir] [...]
_jb_pytest_runner.py: error: unrecognized arguments: --dina
  inifile: None
  rootdir: /Users/mordab/PycharmProjects_BitBucket/orion-tests-core/testsuite/oni/bulk`

**Comment by RonnyPfannschmidt** [comment]
its unclear what you are asking for

at first glance it seems like you have many pipelines that have a argument thats no longer there - in that case an addoption to make it known + triggering a warning for the deprecation seems necessary

**Reply by Formartha** [reply]
There are few pipelines that uses the same base logic, to have backward compatibility, we need to ignore options which are added by default for new developments.

an example is that default pipeline is using --dina argument but the previous version doesn't have this option in, so the test framework will break as shown above.

Hope I better explained the usecase, Ronny.

**Reply by RonnyPfannschmidt** [reply]
based on that description - you need to add a noop/warning to the old framework for that option
there is no option to randomly ignore broken input

you have well specified new input, so just add the code to handle it by just using addoption

**Comment by MohamedAymenGhribi** [comment]
I think that this should be accomplished by change the implementation of pytest_cmdline_parse hook: However, I did not succeed to find a way to do that, seems related to this https://github.com/pytest-dev/pytest/issues/2616
@ceridwen , @nicoddemus , is is possible to help us ?

**Reply by RonnyPfannschmidt** [reply]
again - all thats needed is a conditional addition of the option to warn and ignore - no need for a utter hack

**Reply by MohamedAymenGhribi** [reply]
thanks
is that mean that it is not possible to make changes to pytest_cmdline_parse hook ?

**Comment by RonnyPfannschmidt** [comment]
closing this as its resolvable by creating a backward compat module which is well within the scope of managing ci pipeliens
